go/vt/sqlparser/token.go

```diff
@@ -519,11 +519,17 @@ func (tkn *Tokenizer) scanBindVar() (int, []byte) {
 		tkn.next()
 	}
 	if !isLetter(tkn.lastChar) {
-		// If there isn't a previous error, then return the colon as it may be a valid token
-		if tkn.LastError == nil {
-			return int(':'), buffer.Bytes()
+		switch tkn.lastChar {
```

Comment:
I would recommend renaming this function... I was confused when I saw the name `scanBindVar` since this doesn't have to do with bindvars. Makes sense that bindvars were the only place we parsed colons before, and when I dug through the code to see how this was called I figured it out. Now that we support this assignment operator, renaming this function to something like `scanColonPrefixToken` would be a little more accurate and probably help future code readers understand the function a little faster.